iavl/iavl.py

```diff
@@ -130,12 +130,16 @@ def delete_version(self, v: int) -> int:
         from .diff import diff_tree
 
         counter = 0
+        # optimize the case when deleting from the beginning,
+        # in that case, the previous version will always be 0.
```

Comment:
so we cant add test for case like version 2 node in version 5, deleted in version 7, del when more than 3 versions diff?

Reply:
yeah, we should have more tests for pruning.